tkey/src/main/java/com/web3auth/tkey/ThresholdKey/Modules/TSSModule.java

```diff
@@ -0,0 +1,472 @@
+        String extendedVerifierId = getExtendedVerifier(thresholdKey);
+        String[] split = extendedVerifierId.split("\u001c");
+        String extendedVerifierIdFormatted = split[1] + "\u0015" + tssTag + "\u0016" + nonce;
+
+        VerifierArgs verifierArgs = new VerifierArgs(
+            split[0],
+            split[1],
+            extendedVerifierIdFormatted
+        );
+        TorusPublicKey result = torusUtils.getPublicAddress(nodeDetails.getTorusNodeEndpoints(), nodeDetails.getTorusNodePub(), verifierArgs).get();
+
+        String x = result.getFinalKeyData().getX();
+        String y = result.getFinalKeyData().getY();
+        List<BigInteger> nodeIndexes = result.getNodesData().getNodeIndexes();
+        GetTSSPubKeyResult.Point pubKey = new GetTSSPubKeyResult.Point(x, y);
+        return new GetTSSPubKeyResult(pubKey, nodeIndexes);
+    }
+
+    public static Pair<String, String> getTSSShare(ThresholdKey thresholdKey, String TSSTag, String factorKey, int threshold) throws RuntimeError {
+        if(factorKey.length() > 64) {
+            throw new RuntimeException("Invalid factor Key");
+        }
+        setTSSTag(thresholdKey, TSSTag);
+
+        RuntimeError error = new RuntimeError();
+        String result = jniTSSModuleGetTSSShare(thresholdKey, factorKey, threshold, thresholdKey.curveN, error);
+        String[] splitString = result.split(",", 3);
+
+        if (error.code != 0) {
+            throw error;
+        }
+        return new Pair<>(splitString[0], splitString[1]);
+    }
+
+    public static int getTSSNonce(ThresholdKey thresholdKey, String tssTag, Boolean preFetch) throws RuntimeError {
+        RuntimeError error = new RuntimeError();
+        int nonce = jniGetTSSNonce(thresholdKey, tssTag, error);
+        if (error.code != 0) {
+            throw error;
+        }
+        if (preFetch) {
+            nonce = nonce + 1;
+        }
+        return nonce;
+    }
+
+    public static void copyFactorPub(ThresholdKey thresholdKey, String TSSTag, String factorKey, String newFactorPub, int TSSIndex, ThresholdKeyCallback<Boolean> callback) {
+        if (factorKey.length() > 66) {
+            throw new RuntimeException("Invalid factor Key");
+        }
+        setTSSTag(thresholdKey, TSSTag, result -> {
+            if (result instanceof Result.Error) {
+                throw new RuntimeException("failed to copyFactorPub");
+            }
+        });
+
+        thresholdKey.executor.execute(() -> {
+            try {
+                Result<Boolean> result = copyFactorPub(thresholdKey, factorKey, newFactorPub, TSSIndex);
+                callback.onComplete(result);
+            } catch (Exception e) {
+                Result<Boolean> error = new Result.Error<>(e);
+                callback.onComplete(error);
+            }
+        });
```

Comment:
This should only be run in the callback of setTssTag, since we need to check that it succeeds first. Please check for other instances of this.